### DISCUSSION TITLE
Tasmota does not start with Ethernet cable connected on boot and WiFi disabled

### DISCUSSION THREAD

**Discussion by phoenix83** [arendst/Tasmota#22114] [Support request] [answered]
Hi all, been banging my head against the wall for a couple of days, hopefully someone can provide some padding :)
Similar to #21618 but with different trigger conditions.

Essential details:
- Pre-compiled tasmota32.factory.bin image 14.2.0, flashed with web installer
- WT32-ETH01 clone (my error on AliExpress) uses ESP32-WROOM-32
- Ethernet network is isolated (small switch with WT32-ETH01 and laptop). No access to NTP etc.

Settings changed from default:
```
SO1 0
Template {"NAME":"CueLight","GPIO":[0,1,1,1,419,418,0,0,1,0,160,161,5536,544,5600,0,0,0,0,5568,0,0,0,0,0,0,0,0,416,417,0,1,162,0,0,163],"FLAG":0,"BASE":1}
EthAddress 1
EthClockMode 0
Module 0
SaveData 0
SO0 0
PowerOnState 0
SO53 1
SO146 1
SO3 0
DeviceName CueLight 3
EthIPAddress 10.0.4.3
EthSubnetmask 255.255.255.0
Hostname cuelight3
SO13 1
SO20 1
SO114 1
SwitchMode0 2
SO68 1
```

Once all the above is set up and verified, I disable Wifi (Wifi 0) and still have connectivity via Ethernet, but after reboot I can ping the IP but get no HTTP response.
Looking at the serial log, the boot process seems to be waiting for the action of the Ethernet cable being connected. However it looks like if this happens before the "reset" log entry, whatever process is waiting does not detect it.

Logs:
```
// Power up with no Ethernet cable connected:

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:184
load:0x40078000,len:13228
ho 0 tail 12 room 4
load:0x40080400,len:4
load:0x40080404,len:2924
entry 0x40080570

00:00:00.001 HDW: ESP32-D0WD-V3 v3.1
00:00:00.034 UFS: FlashFS mounted with 4404 kB free
00:00:00.063 CFG: Loaded from File, Count 21
00:00:00.075 QPC: Count 2
00:00:00.084 ROT: Mode 1
00:00:00.090 BRY: GC from 3692 to 3492 bytes, objects freed 4/20 (in 0 ms) - slots from 36/61 to 28/61
00:00:00.118 CFG: No '*.autoconf' file found
00:00:00.123 BRY: GC from 5273 to 3764 bytes, objects freed 8/35 (in 0 ms) - slots from 54/76 to 36/76
00:00:00.123 BRY: Berry initialized, RAM used 3764 bytes
00:00:00.144 BRY: No 'preinit.be'
00:00:00.146 SRC: Restart
00:00:00.152 Project tasmota - CueLight 3 Version 14.2.0(release-tasmota32)-3_0_4(2024-08-14T12:39:45)
00:00:04.112 ETH: Attempting connection...
00:00:04.199 BRY: No 'autoexec.be'
00:00:11.227 QPC: Reset
00:00:13.218 APP: Boot Count 12

// Connect Ethernet cable

00:00:38.114 ETH: Connected at 100Mbps Full Duplex, Mac D0:EF:76:EE:50:13, Hostname cuelight3-eth
00:00:38.115 ETH: IPv4 10.0.4.3, mask 255.255.255.0, gateway 0.0.0.0
00:00:38.235 HTP: Web server active on cuelight3-eth with IP address 10.0.4.3
00:00:38.238 RSL: INFO1 = {"Info1":{"Module":"CueLight","Version":"14.2.0(release-tasmota32)","FallbackTopic":"cmnd/DVES_EE5010_fb/","GroupTopic":"cmnd/tasmotas/"}}
00:00:38.250 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Ethernet":{"Hostname":"cuelight3-eth","IPAddress":"10.0.4.3"}}}
00:00:38.263 RSL: INFO3 = {"Info3":{"RestartReason":"Vbat power on reset","BootCount":13}}
00:00:38.274 RSL: RESULT = {"POWER1":"OFF"}
00:00:38.275 RSL: POWER1 = OFF
00:00:38.276 RSL: RESULT = {"POWER2":"OFF"}
00:00:38.277 RSL: POWER2 = OFF
00:00:38.288 RSL: RESULT = {"POWER3":"OFF"}
00:00:38.289 RSL: POWER3 = OFF
00:00:38.290 RSL: RESULT = {"POWER4":"OFF"}
00:00:38.291 RSL: POWER4 = OFF
00:00:38.297 TFS: File 'autoexec.bat' not found
00:00:40.077 ETH: IPv6 Local fe80::d2ef:76ff:feee:5013
00:00:42.207 BRY: GC from 5282 to 4054 bytes, objects freed 15/44 (in 1 ms) - slots from 83/91 to 49/91
00:00:42.209 RSL: STATE = {"Time":"1970-01-01T00:00:42","Uptime":"0T00:00:39","UptimeSec":39,"Heap":168,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":44},"POWER1":"OFF","Channel1":100,"POWER2":"OFF","Channel2":100,"POWER3":"OFF","Channel3":100,"POWER4":"OFF","Channel4":100,"Color":"FFFFFFFF"}
00:00:42.245 RSL: SENSOR = {"Time":"1970-01-01T00:00:42","Switch1":"OFF","Switch2":"OFF","Switch3":"OFF","Switch4":"OFF","ESP32":{"Temperature":42.2},"TempUnit":"C"}

// restart 1 from GUI console

00:02:18.666 SRC: WebConsole from 10.0.4.83
00:02:18.666 CMD: Grp 0, Cmd 'RESTART', Idx 1, Len 1, Pld 1, Data '1'
00:02:18.667 RSL: RESULT = {"Restart":"Restarting"}
00:02:18.869 CFG: Saved, Count 22, Bytes 4096
00:02:19.725 APP: Restarting
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:184
load:0x40078000,len:13228
ho 0 tail 12 room 4
load:0x40080400,len:4
load:0x40080404,len:2924
entry 0x40080570

00:00:00.002 HDW: ESP32-D0WD-V3 v3.1
00:00:00.039 UFS: FlashFS mounted with 4404 kB free
00:00:00.067 CFG: Loaded from File, Count 22
00:00:00.078 QPC: Count 1
00:00:00.087 ROT: Mode 1
00:00:00.093 BRY: GC from 3692 to 3492 bytes, objects freed 4/20 (in 0 ms) - slots from 36/61 to 28/61
00:00:00.122 CFG: No '*.autoconf' file found
00:00:00.126 BRY: GC from 5273 to 3764 bytes, objects freed 8/35 (in 1 ms) - slots from 54/76 to 36/76
00:00:00.127 BRY: Berry initialized, RAM used 3764 bytes
00:00:00.148 BRY: No 'preinit.be'
00:00:00.153 Project tasmota - CueLight 3 Version 14.2.0(release-tasmota32)-3_0_4(2024-08-14T12:39:45)
00:00:01.711 ETH: Attempting connection...
00:00:01.713 ETH: Connected at 100Mbps Full Duplex, Mac D0:EF:76:EE:50:13, Hostname cuelight3-eth
00:00:01.800 BRY: No 'autoexec.be'
00:00:03.076 ETH: IPv6 Local fe80::d2ef:76ff:feee:5013
00:00:07.926 QPC: Reset
00:00:09.917 APP: Boot Count 13

// Disconnect Ethernet cable

00:00:45.713 ETH: Disconnected

// Connect Ethernet cable

00:00:55.713 ETH: Connected at 100Mbps Full Duplex, Mac D0:EF:76:EE:50:13, Hostname cuelight3-eth
00:00:55.714 ETH: IPv4 10.0.4.3, mask 255.255.255.0, gateway 0.0.0.0
00:00:55.935 HTP: Web server active on cuelight3-eth with IP address 10.0.4.3
00:00:55.938 RSL: INFO1 = {"Info1":{"Module":"CueLight","Version":"14.2.0(release-tasmota32)","FallbackTopic":"cmnd/DVES_EE5010_fb/","GroupTopic":"cmnd/tasmotas/"}}
00:00:55.951 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Ethernet":{"Hostname":"cuelight3-eth","IPAddress":"10.0.4.3"}}}
00:00:55.963 RSL: INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":14}}
00:00:55.975 RSL: RESULT = {"POWER1":"OFF"}
00:00:55.976 RSL: POWER1 = OFF
00:00:55.977 RSL: RESULT = {"POWER2":"OFF"}
00:00:55.977 RSL: POWER2 = OFF
00:00:55.989 RSL: RESULT = {"POWER3":"OFF"}
00:00:55.990 RSL: POWER3 = OFF
00:00:55.991 RSL: RESULT = {"POWER4":"OFF"}
00:00:55.991 RSL: POWER4 = OFF
00:00:55.997 TFS: File 'autoexec.bat' not found
00:00:57.077 ETH: IPv6 Local fe80::d2ef:76ff:feee:5013
00:00:59.907 BRY: GC from 5290 to 4054 bytes, objects freed 15/44 (in 1 ms) - slots from 82/91 to 49/91
00:00:59.909 RSL: STATE = {"Time":"1970-01-01T00:00:59","Uptime":"0T00:01:00","UptimeSec":60,"Heap":167,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":44},"POWER1":"OFF","Channel1":100,"POWER2":"OFF","Channel2":100,"POWER3":"OFF","Channel3":100,"POWER4":"OFF","Channel4":100,"Color":"FFFFFFFF"}
00:00:59.945 RSL: SENSOR = {"Time":"1970-01-01T00:00:59","Switch1":"OFF","Switch2":"OFF","Switch3":"OFF","Switch4":"OFF","ESP32":{"Temperature":42.8},"TempUnit":"C"}
```

The workaround suggested in #21618 (Rule1 ON System#Init DO teleperiod ENDON) does not work:
```
00:00:00.002 HDW: ESP32-D0WD-V3 v3.1
00:00:00.006 UFS: FlashFS mounted with 4404 kB free
00:00:00.011 CFG: Loaded from File, Count 23
00:00:00.022 QPC: Count 1
00:00:00.031 ROT: Mode 1
00:00:00.037 BRY: GC from 3692 to 3492 bytes, objects freed 4/20 (in 0 ms) - slots from 36/61 to 28/61
00:00:00.045 CFG: No '*.autoconf' file found
00:00:00.049 BRY: GC from 5273 to 3764 bytes, objects freed 8/35 (in 1 ms) - slots from 54/76 to 36/76
00:00:00.050 BRY: Berry initialized, RAM used 3764 bytes
00:00:00.064 BRY: No 'preinit.be'
00:00:00.074 Project tasmota - CueLight 3 Version 14.2.0(release-tasmota32)-3_0_4(2024-08-14T12:39:45)
00:00:01.688 ETH: Attempting connection...
00:00:01.689 ETH: Connected at 100Mbps Full Duplex, Mac D0:EF:76:EE:50:13, Hostname cuelight3-eth
00:00:01.749 BRY: No 'autoexec.be'
00:00:01.846 RUL: SYSTEM#INIT performs 'teleperiod'
00:00:01.846 SRC: Rule
00:00:01.847 CMD: Grp 0, Cmd 'TELEPERIOD', Idx 1, Len 0, Pld -99, Data ''
00:00:01.858 RSL: RESULT = {"TelePeriod":300}
00:00:01.995 BRY: GC from 4594 to 3902 bytes, objects freed 13/38 (in 0 ms) - slots from 72/91 to 41/76
00:00:01.997 RSL: STATE = {"Time":"1970-01-01T00:00:01","Uptime":"0T00:00:02","UptimeSec":2,"Heap":175,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":3,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":38},"POWER1":"OFF","Channel1":100,"POWER2":"OFF","Channel2":100,"POWER3":"OFF","Channel3":100,"POWER4":"OFF","Channel4":100,"Color":"FFFFFFFF"}
00:00:02.033 RSL: SENSOR = {"Time":"1970-01-01T00:00:02","Switch1":"OFF","Switch2":"OFF","Switch3":"OFF","Switch4":"OFF","ESP32":{"Temperature":42.8},"TempUnit":"C"}
00:00:03.053 ETH: IPv6 Local fe80::d2ef:76ff:feee:5013
00:00:07.990 QPC: Reset
00:00:09.981 APP: Boot Count 14
```

**Comment by Noschvie** [comment]
How is the setting for `EthType `on your device?

**Reply by phoenix83** [reply]
`EthType` is 0 (default). The chip is marked 8720A and the Ethernet does work so I think that is correct.

**Reply by Noschvie** [reply]
Have a ESP32-D0WD **v1.1** running with ETH and it works fine.
Seems to be related to v1.1 vs v3.1 ?

```
00:00:00.002 HDW: ESP32-D0WD v1.1 
00:00:00.033 UFS: FlashFS mounted with 304 kB free
00:00:00.056 CFG: Loaded from File, Count 58
00:00:00.243 QPC: Count 1
00:00:00.248 I2C: Bus1 using GPIO01(SCL) and GPIO03(SDA)
00:00:00.297 BRY: Berry initialized, RAM used 3764 bytes
00:00:00.310 Project tasmota - Tasmota Ethernet 5944 Version 14.2.0(release-tasmota32)-3_0_4(2024-08-14T12:39:45)
00:00:01.703 ETH: Connected at 100Mbps Full Duplex, Mac 24:DC:C3:C7:F7:3B, Hostname tasmota-C7F738-5944-eth
00:00:01.880 I2C: VL53L0X found at 0x29
10:51:34.414 HTP: Web server active on tasmota-C7F738-5944-eth with IP address 192.168.7.178
10:51:35.441 MQT: Attempting connection...
10:51:35.609 MQT: Connected
10:51:35.610 MQT: tele/tasmota_C7F738/LWT = Online (retained)
10:51:35.612 MQT: cmnd/tasmota_C7F738/POWER = 
10:51:35.619 MQT: tele/tasmota_C7F738/INFO1 = {"Info1":{"Module":"WT32_ETH01","Version":"14.2.0(release-tasmota32)","FallbackTopic":"cmnd/DVES_C7F738_fb/","GroupTopic":"cmnd/tasmotas/"}}
10:51:35.632 MQT: tele/tasmota_C7F738/INFO2 = {"Info2":{"WebServerMode":"Admin","Ethernet":{"Hostname":"tasmota-C7F738-5944-eth","IPAddress":"192.168.7.178"}}}
10:51:35.646 MQT: tele/tasmota_C7F738/INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":26}}
10:51:36.822 MQT: stat/tasmota_C7F738/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota32/release/tasmota32.bin","RestartReason":"Software reset CPU","Uptime":"0T00:00:03","StartupUTC":"2024-09-12T09:51:33","Sleep":50,"CfgHolder":4617,"BootCount":25,"BCResetTime":"2024-02-29T20:40:56","SaveCount":59}}
10:51:36.924 MQT: stat/tasmota_C7F738/STATUS10 = {"StatusSNS":{"Time":"2024-09-12T10:51:36","VL53L0X":{"Distance":9.1}}}
10:51:36.973 MQT: stat/tasmota_C7F738/STATUS11 = {"StatusSTS":{"Time":"2024-09-12T10:51:36","Uptime":"0T00:00:03","UptimeSec":3,"Heap":161,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":23,"MqttCount":1,"Berry":{"HeapUsed":3,"Objects":45}}}
10:51:39.979 MQT: tele/tasmota_C7F738/STATE = {"Time":"2024-09-12T10:51:39","Uptime":"0T00:00:06","UptimeSec":6,"Heap":163,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Berry":{"HeapUsed":3,"Objects":45}}
10:51:39.995 MQT: tele/tasmota_C7F738/SENSOR = {"Time":"2024-09-12T10:51:39","VL53L0X":{"Distance":9.1}}
10:51:40.985 QPC: Reset
10:52:01.110 CMD: wifi
10:52:01.113 MQT: stat/tasmota_C7F738/RESULT = {"Wifi":"OFF","Mode":"low rate"}
10:52:06.312 CMD: wificonfig 
10:52:06.315 MQT: stat/tasmota_C7F738/RESULT = {"WifiConfig":{"4":"Retry"}}
10:52:09.513 CMD: wifi
```

**Comment by phoenix83** [comment]
I just tried with 13.0 with the same result. Kept config to a minimum:
```
BackLog SO1 0; Template {"NAME":"CueLight","GPIO":[0,1,1,1,419,418,0,0,1,0,160,161,5536,544,5600,0,0,0,0,5568,0,0,0,0,0,0,0,0,416,417,0,1,162,0,0,163],"FLAG":0,"BASE":1}; EthAddress 1; EthClockMode 0; Module 0
BackLog DeviceName CueLight 3; EthIPAddress 10.0.4.3; EthSubnetmask 255.255.255.0; Hostname cuelight3
```

**Comment by s-hadinger** [comment]
I will look at your logs. 

However I'm really surprised. I have several Tasmota devices that are using Ethernet only (wifi disabled) and the web UI works fine.

**Comment by s-hadinger** [comment]
Oh, the main different I see is that you have a static IP address.

What's obvious in the logs is that when you have only Ethernet active, the static IP address is not assigned.
```
00:00:01.689 ETH: Connected at 100Mbps Full Duplex, Mac D0:EF:76:EE:50:13, Hostname cuelight3-eth
<--- absence of IPv4 log
00:00:01.749 BRY: No 'autoexec.be'
```

whereas when it works, the IP address is logged 1 ms after the connection
```
00:00:38.114 ETH: Connected at 100Mbps Full Duplex, Mac D0:EF:76:EE:50:13, Hostname cuelight3-eth
00:00:38.115 ETH: IPv4 10.0.4.3, mask 255.255.255.0, gateway 0.0.0.0
```

I don't know the reason, but it narrows down the problem to static IP addresses

**Comment by phoenix83** [comment, marked as answer]
Thanks @s-hadinger, your discovery set me on the right track to resolving it.

I did some playing with DHCP and saw the same as you: reboot with Ethernet connected worked fine with DHCP. However I also noticed that after it had received a DHCP lease once, changing back to fixed IP and rebooting would work fine (boot would complete). Couple this with status 5 showing a default gateway IP (192.168.1.1) for the (disabled) wifi interface triggered the 💡 moment.

In summary, an Ethernet gateway IP must be configured to get around this. With that set (even though it doesn't exist) Tasmota boots fine with the Ethernet cable connected.
